File: leetcode3/이진희/3870_Count Commas in Range.java
/*

1. 아이디어 : 콤마가 생기는 기준점(1,000, 1,000,000 등)을 초과하는 숫자의 개수를 누적해서 더함

2. 시간복잡도 : O(logN)

3. 자료구조/알고리즘 : 단순계산

 */

class Solution {
    public int countCommas(int n) {
        long totalCommas = 0;
        
        for (long i = 1000; i <= n; i *= 1000) {
            totalCommas += (n - i + 1);
        }
        
        return (int)totalCommas;
    }
}

File: leetcode3/최원준/3870. Count Commas in Range.py
#

'''
1. 아이디어 :
n은 100,000 까지.
1부터 999까지는 0개, 그 외에는 1개 이므로, n - 999

2. 시간복잡도 :
    O(1)

3. 자료구조/알고리즘 :
-

'''

class Solution:
    def countCommas(self, n: int) -> int:
        return max(0, n - 999)